pypesto/optimize/optimizer.py

```
@@ -691,11 +691,7 @@ class DlibOptimizer(Optimizer):
     def __init__(self, options: dict = None):
         super().__init__()
 
-        self.options = options
-        if self.options is None:
-            self.options = DlibOptimizer.get_default_options(self)
-        elif "maxiter" not in self.options:
-            raise KeyError("Dlib options are missing the keyword maxiter.")
+        self.options = self.extend_options_with_defaults(options)
```

Hm, it wasn't there before either - but maybe raise on any unsupported options? Currently, exactly two are supported. Would be good to alert the user if they, for example, misspell `maxiter` as `max_iter`.

---

Would agree and also in favor for raising errors, if users missspecify anything.